zigpy/quirks/registry.py

```diff
@@ -19,11 +19,16 @@
 SIG_NODE_DESC = "node_desc"
 SIG_SKIP_CONFIG = "skip_configuration"
 
+TYPE_MODEL_QUIRKS_LIST = Dict[str, List["zigpy.quirks.CustomDevice"]]
+TYPE_MANUF_QUIRKS_DICT = Dict[str, TYPE_MODEL_QUIRKS_LIST]
+
 
 class DeviceRegistry:
     def __init__(self, *args, **kwargs):
-        dd = collections.defaultdict
-        self._registry = dd(lambda: dd(list))
+        self._model_quirks: TYPE_MODEL_QUIRKS_LIST = collections.defaultdict(list)
+        self._registry: TYPE_MANUF_QUIRKS_DICT = collections.defaultdict(
+            lambda: self._model_quirks
```

Comment:
This means the same instance (models list) will be shared for all the manufacturer ids. And is the cause of some issues people are having with device matching, eg: https://github.com/zigpy/zha-device-handlers/pull/538#issuecomment-738039431

Reply:
I need to test but you are probably right. If you reverse it, does it match correctly then?

Reply:
Yes, it matches correctly when i change it to `lambda: collections.defaultdict(list)`

Reply:
ok, that would require a change to zha-device-handlers too. Should do it before next release is shipped

Reply:
It is now in 1.0.0b3 and should be in the next release. Give it a try and open an issue if still is not fixed.

Reply:
confirmed working, thanks.